maven.grammar/src/org/netbeans/modules/maven/hyperlinks/HyperlinkProviderImpl.java

```diff
@@ -453,5 +215,331 @@ private FileObject getPath(FileObject parent, String path) {
+                                        xml.moveNext(); 
+                                        xml.moveNext();
+                                        token = xml.token();
+                                        if (token.text().toString().startsWith("${")) { //NOI18N
+                                            Project nbprj = getProject(doc);
+                                            try { 
+                                                version = (String)PluginPropertyUtils.createEvaluator(nbprj).evaluate(token.text().toString());
+                                            } catch (ExpressionEvaluationException eee) {
+                                                LOG.log(Level.INFO, "Unable to evaluate property: " + token.text().toString(), eee);
+                                            }
+                                        } else {
+                                            version = token.text().toString();
+                                        }
+                                        xml.moveNext();
+                                    } else if (TokenUtilities.equals(token.text(), "<type")) { //NOI18N
+                                        xml.moveNext();
+                                        xml.moveNext();
+                                        token = xml.token();
+                                        type = token.text().toString();
+                                        xml.moveNext();
+                                    }
+                                }
+                                // handle cases where the version element is covered in a 
+                                // parent pom/dependenciesManagement
+                                if (version == null) {
+                                    MavenProject mavenProject = getNbMavenProject(doc).getMavenProject();
+                                    for (Artifact artifact : mavenProject.getArtifacts()) {
+                                        if (artifact.getGroupId().equals(groupId) 
+                                                && artifact.getArtifactId().equals(artifactId)) {
+                                            version = artifact.getVersion();
+                                            break;
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }     
+        }
+        
+        String[] getTooltipText() {
+            if (isText) {
+                //we are in element text
+                String text = ftext;
+                int tokenOff = ftokenOff;
+                if (isMavenProperty()){
+                    Tuple tup = findProperty(text, tokenOff, documentOffset);
+
+                    if (tup != null) {
+                        String prop = tup.value.substring("${".length(), tup.value.length() - 1); //remove the brackets
```

Comment:
The compiler will not optimize the length() call to a constant.

Reply:
I have no idea why I wrote that instead of at least 2 :(